series/type-time.go

```diff
@@ -0,0 +1,150 @@
+package series
+
+import (
+	"fmt"
+	"time"
+)
+
+// timeElement is the concrete implementation of the Element interface for
+// time.Time. If the stored time.Time is zero, it will be considered as a NaN
+// element.
```

Comment:
Since the zero time will be considered NaN, there is no need to have a separate `nan` field on the struct. Furthermore, after removing it the rest of the code should be adjusted to match the change.

Maybe we could do `type timeElement time.Time` instead, since we only need one field